src/Evolution/Systems/GrMhd/ValenciaDivClean/TimeDerivativeTerms.cpp

```diff
@@ -0,0 +1,200 @@
+    const gsl::not_null<Scalar<DataVector>*> one_over_w_squared,
+    const gsl::not_null<Scalar<DataVector>*> pressure_star,
+    const gsl::not_null<Scalar<DataVector>*>
+        pressure_star_lapse_sqrt_det_spatial_metric,
+    const gsl::not_null<Scalar<DataVector>*> transport_velocity,
+
+    const gsl::not_null<tnsr::I<DataVector, 3, Frame::Inertial>*> tilde_s_up,
+    const gsl::not_null<tnsr::II<DataVector, 3, Frame::Inertial>*>
+        densitized_stress,
+    const gsl::not_null<tnsr::ijj<DataVector, 3, Frame::Inertial>*>
+        spatial_christoffel_first_kind,
+    const gsl::not_null<tnsr::Ijj<DataVector, 3, Frame::Inertial>*>
+        spatial_christoffel_second_kind,
+    const gsl::not_null<tnsr::I<DataVector, 3, Frame::Inertial>*>
+        trace_spatial_christoffel_second,
+    const gsl::not_null<Scalar<DataVector>*> h_rho_w_squared_plus_b_squared,
+
+    const Scalar<DataVector>& tilde_d, const Scalar<DataVector>& tilde_tau,
+    const tnsr::i<DataVector, 3, Frame::Inertial>& tilde_s,
+    const tnsr::I<DataVector, 3, Frame::Inertial>& tilde_b,
+    const Scalar<DataVector>& tilde_phi, const Scalar<DataVector>& lapse,
+    const tnsr::I<DataVector, 3, Frame::Inertial>& shift,
+    const Scalar<DataVector>& sqrt_det_spatial_metric,
+    const tnsr::ii<DataVector, 3, Frame::Inertial>& spatial_metric,
+    const tnsr::II<DataVector, 3, Frame::Inertial>& inv_spatial_metric,
+    const tnsr::i<DataVector, 3, Frame::Inertial>& d_lapse,
+    const tnsr::iJ<DataVector, 3, Frame::Inertial>& d_shift,
+    const tnsr::ijj<DataVector, 3, Frame::Inertial>& d_spatial_metric,
+    const Scalar<DataVector>& pressure,
+    const tnsr::I<DataVector, 3, Frame::Inertial>& spatial_velocity,
+    const Scalar<DataVector>& lorentz_factor,
+    const tnsr::I<DataVector, 3, Frame::Inertial>& magnetic_field,
+
+    const Scalar<DataVector>& rest_mass_density,
+    const Scalar<DataVector>& specific_enthalpy,
+    const tnsr::ii<DataVector, 3, Frame::Inertial>& extrinsic_curvature,
+    const double constraint_damping_parameter) noexcept {
+  raise_or_lower_index(spatial_velocity_one_form, spatial_velocity,
+                       spatial_metric);
+  raise_or_lower_index(magnetic_field_one_form, magnetic_field, spatial_metric);
+  dot_product(magnetic_field_dot_spatial_velocity, magnetic_field,
+              *spatial_velocity_one_form);
+  dot_product(magnetic_field_squared, magnetic_field, *magnetic_field_one_form);
+  get(*one_over_w_squared) = 1.0 / square(get(lorentz_factor));
+  get(*pressure_star) =
+      get(pressure) + 0.5 * square(get(*magnetic_field_dot_spatial_velocity)) +
+      0.5 * get(*magnetic_field_squared) * get(*one_over_w_squared);
+  get(*pressure_star_lapse_sqrt_det_spatial_metric) =
+      get(sqrt_det_spatial_metric) * get(lapse) * get(*pressure_star);
+
+  // lapse b_i / W = lapse (B_i / W^2 + v_i (B^m v_m)
+  // reuse allocation for spatial velocity one form
```

Comment:
do we still want to do this?  It means the `spatial_velocity_one_form` is not correct exiting the function...

Reply:
yea, let's not do that... I'll add another temporary tag (cheap anyway since it's all one allocation) and then we can safely reuse things. Good catch!